docs/proposals/refactor-keadm-on-the-cloud.md

```diff
@@ -0,0 +1,283 @@
+
+```flow
+start=>start: keadm init-beta --set cloudCore.modules.cloudHub.advertiseAddress[0]=192.168.88.6 --profile iptablesMgrMode=external
+keadm init-beta --set cloudCore.modules.cloudHub.advertiseAddress[0]=192.168.88.6 --profile version=1.9.0
+
+render=>operation: accept the key-value pairs, apply the custom charts to the cluster
+apply=>end: deployed
+
+start->render->apply
+```
+
+> Refer to the above configuration in Plan A for the profile parameters
+
+## Referenced command line arguments for Plan B
+
+### Init-beta
+
+> The former keadm init cmd will keep several versions, later, the init-beta cmd can be replaced by init.
+
+This cmd is used for installing the core components, such as cloudcore, external iptableMgr, edgemesh, cloudcore hamode, etc. 
+
+
+#### --profile
+
+version=1.9.0
+> the given version will seek the recommended parameters to install
+
+iptablesMgrMode=external
+> the given version will seek the recommended parameters to install
+
+enableCloudCoreHaMode=true
+> enable ha mode for cloudcore
+
+enableEdgeMesh=true
+> enable edgemesh component
+
+#### --set
+
+A list of set flags like helm flags.
+
+#### --charts
+Allow appending file directories of charts to keadm, separated by commas
+
+##### --files, -f, --manifests
+Allow appending file paths of manifests to keadm, separated by commas
+
+#### --namespace
+
+> The namespace to install, default is kubeedge.
+
+#### --dry-run
+
+> print the generated k8s resources on the stdout, not actual excute. Always use in debug mode.
+
+### config
+
+Like kubeadm config images list/pull
+
+```
+keadm config images list/pull
+```
+### manifest
+
+Also support the "--set" list above for custom modification.
+
+#### generate
+
+Generate the k8s resources, and can combine into one file like the cmd "kubectl kustomize".
+
+```
+keadm manifest generate <your original installation options> | kubectl delete -f -
+```
+
+##### --kustomize
+will generate all the files to one file, by default will be separated.
+
+##### --files, -f
+Allow appending file paths of manifests to keadm, separated by commas
+
+#### --charts
+Allow appending file directories of charts to keadm, separated by commas
+
+### reset-beta
+
+> Uninstall the exsited helm charts or manifests on the cloud.
+
+#### --delete-namespace, -D
+
+> This flag will forced-delete all resources and the namespace. 
+
+#### --grace-period
+
+> All resources will be gracefully removed after the interval.
+
+### profile
+
+```
+keadm profile list
+> iptablesMgrMode
+> version
+> enableCloudCoreHaMode
+> enableEdgeMesh
+```
+
+## Sub tasks for plan B
+
+- Support haproxy for helm
+- keadm  init-beta
+- keadm config images list/pull
+- keadm reset-beta
+- keadm manifest  > k8s resources , kubectl  kustomize 
```

Comment:
Do we need to support `kustomize`?
I think it is enough to generate the k8s resource manifest according to the default configuration, if need to modify it directly edit the yaml.

Reply:
Flag --kustomize will combine all resources into one file. Seems to be more simple.  I think it might help that it's not urgent. :)

Reply:
`kubectl` supports the execution of the entire directory. In fact, I don’t think there will be a scene that requires a single yaml, and it will be cumbersome to modify.

Reply:
What do you think? /cc @fisherxu

Reply:
Temporarily commented out.